test/infrastructure/docker/api/v1beta2/devcluster_types.go

```diff
@@ -119,24 +119,35 @@ type DevClusterStatus struct {
 	// +optional
 	FailureDomains clusterv1beta1.FailureDomains `json:"failureDomains,omitempty"`
 
-	// conditions defines current service state of the DevCluster.
+	// conditions represents the observations of a DevCluster's current state.
 	// +optional
-	Conditions clusterv1.Conditions `json:"conditions,omitempty"`
+	// +listType=map
+	// +listMapKey=type
+	// +kubebuilder:validation:MaxItems=32
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
 
-	// v1beta2 groups all the fields that will be added or modified in DevCluster's status with the V1Beta2 version.
+	// deprecated groups all the status fields that are deprecated and will be removed when support for v1beta1 will be dropped.
```

Comment:
nit In order to distinguish this from deprecated.v1beta1 we used 
```suggestion
	// deprecated groups all the status fields that are deprecated and will be removed when all the nested field are removed..
```
(same for the struct + other types)